nautobot_ssot/metrics.py

```diff
@@ -0,0 +1,156 @@
+"""Nautobot SSoT framework level metrics."""
+from django.conf import settings
+from prometheus_client.core import GaugeMetricFamily
+from nautobot.extras.models.jobs import Job
+from nautobot_ssot.models import Sync
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_ssot", {})
+
+
+def metric_ssot_jobs():
+    """Extracts duration of latest SSoT Job run.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    ssot_job_durations = GaugeMetricFamily(
+        "nautobot_ssot_duration_microseconds",
+        "Nautobot SSoT Job Phase Duration in microseconds",
+        labels=["phase", "job"],
```

Comment:
The unit should be seconds (see [here](https://prometheus.io/docs/practices/naming/#base-units) for a list of base units).

Reply:
The sample is a float value, so if you pass in a float you don't lose any precision.

Reply:
Strange. When I was testing this and used the seconds attribute it appeared to be returned as an int not a float. Is there a specific way to get it as a float? That's why I went to microseconds originally.

Reply:
Oh sorry, misunderstanding - my point was to get the value in microseconds and then divide it by a 1000 and pass that to the metric.